### DISCUSSION TITLE
400 Bad Request within express.js app running tRPC routes ... "No input expected"

### DISCUSSION THREAD

**Discussion by michealroberts** [trpc/trpc#2390] [Q&A]
I have the following route defined:

```ts
export const telescopesRouter = createRouter().query('all', {
  async resolve() {
    return findManyTelescopes(prisma)
  },
})
```

Which works and returns a 200 when the express server is running locally.

However, I see the following which to me is a slight issue:

<img width="635" alt="Screenshot 2022-08-07 at 16 06 40" src="https://user-images.githubusercontent.com/84131395/183297618-f6626fbc-33f8-4320-8af9-1a738a3cfc79.png">

The client seems to be passing a strange batch and input query alongside the URL.

However, when this is called in production, I am seeing the following response:

```json
[
  {
    "id": null,
    "error": {
      "message": "No input expected",
      "code": -32600,
      "data": {
        "code": "BAD_REQUEST",
        "httpStatus": 400,
        "path": "telescopes.all"
      }
    }
  }
]
```

Yet also in production, when I strip the URL back to what I am sending to the client, I see the correct response:

```json
{
  "id": null,
  "result": {
    "type": "data",
    "data": [
      {
        "id": "090640c8-4dff-4c3c-b562-d3ee1e4c3bc9",
        "domain": "pleiades.namibia.observerly.com",
        "name": "Namibiascope 1",
        "location": "Namibia",
        "latitude": "-19.566278059",
        "longitude": "16.007009209",
        "elevation": "1290"
      },
      {
        "id": "942dff4e-9038-45a2-a1a1-7247ad578bb8",
        "domain": "pleiades.chile.observerly.com",
        "name": "Chilescope 1",
        "location": "Chile",
        "latitude": "-34.109327109",
        "longitude": "-71.240122387",
        "elevation": "1890"
      }
    ]
  }
}
```

The fact that I am getting a response from the server, and it is acting almost identical to localhost, except when being called from the tRPC build in client ... it has an issue and throws a 400 Bad Request ...

Could this be an issue when putting something behind a k8s cluster / Cloudflare proxy? Do I need to do anything specific to get this to work?

The production server is just a standard node / express app - deployed on k8s cluster behind a Cloudflare proxy. 

Could anyone advise on what is going on here and why this simple request in production is failing?

**Comment by michealroberts** [comment]
I'm happy to give access to the repo for a minimal reproduction of the issue, however, the issue is only occurring within the production environment ... and only when the batch and input are being sent in the URL path from the client ...

**Comment by KATT** [comment]
Looks like you're sending an input to a procedure that don't expect an input.
